### PR TITLE
Add tests for /get_missing_events

### PR DESCRIPTION
**Pull request by S7evinK**
Adds `Inbound federation can return missing events for $vis visibility` from `tests/50federation/33room-get-missing-events.pl`

### CODE REVIEW

File: tests/federation_room_get_missing_events_test.go [hunk 1]
```diff
@@ -314,3 +313,135 @@ func TestOutboundFederationIgnoresMissingEventWithBadJSONForRoomVersion6(t *test
...
+			// * the message
+			events := result.Events.UntrustedEvents(roomVersion)
+			i := 0
+			for _, ev := range events {
+				must.EqualStr(t, ev.RoomID(), roomID, "unexpected roomID")
+				switch i {
+				case 0:
+					must.EqualStr(t, ev.Type(), "m.room.member", "not a membership event")
+					must.EqualStr(t, *ev.StateKey(), alice.UserID, "unexpected creator")
+					i++
```
Comment (DMRobertson) [inline tests/federation_room_get_missing_events_test.go:410]: Would it be easier to loop over `i, ev := range events` here and additionally check that `len(events)` is the correct size?

**Review by DMRobertson — APPROVED** [review]
Seems like a faithful port of the sytest to me. Should we remove it from sytest?

I did have a style suggestion, but happy for this to go-in as is!

File: tests/federation_room_get_missing_events_test.go (hunk 1 above)
Comment (S7evinK) [reply tests/federation_room_get_missing_events_test.go]: I've updated the test. Now we're also checking that we received the message, if the visibility is `shared`

**Review by DMRobertson — APPROVED** [review]
LGTM, thanks!

**Pull request merged**